include/exadg/incompressible_navier_stokes/postprocessor/output_generator.cpp

```diff
@@ -185,75 +189,49 @@ OutputGenerator<dim, Number>::setup(NavierStokesOperator const &    navier_stoke
-      // time step.
-      if(reset_counter)
-      {
-        if(time > output_data.start_time)
-        {
-          output_counter +=
-            int((time - output_data.start_time + EPSILON) / output_data.interval_time);
-        }
-        reset_counter = false;
-      }
-
-      if(time > (output_data.start_time + output_counter * output_data.interval_time - EPSILON))
-      {
-        pcout << std::endl
-              << "OUTPUT << Write data at time t = " << std::scientific << std::setprecision(4)
-              << time << std::endl;
-
-        calculate_additional_fields(velocity, time, time_step_number);
-
-        write_output<dim>(output_data,
-                          *dof_handler_velocity,
-                          *dof_handler_pressure,
-                          *mapping,
-                          velocity,
-                          pressure,
-                          additional_fields,
-                          output_counter,
-                          mpi_comm);
-
-        ++output_counter;
-      }
-    }
-    else
-    {
-      pcout << std::endl
-            << "OUTPUT << Write " << (output_counter == 0 ? "initial" : "solution") << " data"
-            << std::endl;
-
-      calculate_additional_fields(velocity, time, time_step_number);
-
-      write_output<dim>(output_data,
-                        *dof_handler_velocity,
-                        *dof_handler_pressure,
-                        *mapping,
-                        velocity,
-                        pressure,
-                        additional_fields,
-                        output_counter,
-                        mpi_comm);
-
-      ++output_counter;
-    }
+  if(Utilities::is_unsteady_timestep(time_step_number))
+  {
+    pcout << std::endl
+          << "OUTPUT << Write data at time t = " << std::scientific << std::setprecision(4) << time
+          << std::endl;
+
+    calculate_additional_fields(velocity, time, time_step_number);
+
+    write_output<dim>(output_data,
+                      *dof_handler_velocity,
+                      *dof_handler_pressure,
+                      *mapping,
+                      velocity,
+                      pressure,
+                      additional_fields,
+                      time_control.get_counter(),
+                      mpi_comm);
+  }
+  else
+  {
+    pcout << std::endl
+          << "OUTPUT << Write " << (time_control.get_counter() == 0 ? "initial" : "solution")
+          << " data" << std::endl;
+
+    calculate_additional_fields(velocity, time, time_step_number);
+
+    write_output<dim>(output_data,
+                      *dof_handler_velocity,
+                      *dof_handler_pressure,
+                      *mapping,
+                      velocity,
+                      pressure,
+                      additional_fields,
+                      time_control.get_counter(),
+                      mpi_comm);
```

Comment:
Here, we should do the same as in the other comment above, i.e. write the function calls only once for steady/unsteady problems.